diff --git a/linux-3.0.1/include/sound/soc.h b/linux-3.0.1/include/sound/soc.h
index 74921f2..981ed2d 100755
--- a/linux-3.0.1/include/sound/soc.h
+++ b/linux-3.0.1/include/sound/soc.h
@@ -463,14 +463,14 @@ struct snd_soc_codec {
 	/* runtime */
 	struct snd_ac97 *ac97;  /* for ad-hoc ac97 devices */
 	unsigned int active;
-	unsigned int cache_only:1;  /* Suppress writes to hardware */
-	unsigned int cache_sync:1; /* Cache needs to be synced to hardware */
-	unsigned int suspended:1; /* Codec is in suspend PM state */
-	unsigned int probed:1; /* Codec has been probed */
-	unsigned int ac97_registered:1; /* Codec has been AC97 registered */
-	unsigned int ac97_created:1; /* Codec has been created by SoC */
+	unsigned int cache_only:1;       /* Suppress writes to hardware */
+	unsigned int cache_sync:1;       /* Cache needs to be synced to hardware */
+	unsigned int suspended:1;        /* Codec is in suspend PM state */
+	unsigned int probed:1;           /* Codec has been probed */
+	unsigned int ac97_registered:1;  /* Codec has been AC97 registered */
+	unsigned int ac97_created:1;     /* Codec has been created by SoC */
 	unsigned int sysfs_registered:1; /* codec has been sysfs registered */
-	unsigned int cache_init:1; /* codec cache has been initialized */
+	unsigned int cache_init:1;       /* codec cache has been initialized */
 
 	/* codec IO */
 	void *control_data; /* codec control (i2c/3wire) data */
diff --git a/linux-3.0.1/sound/core/control.c b/linux-3.0.1/sound/core/control.c
index 9ce00ed..21308e1 100644
--- a/linux-3.0.1/sound/core/control.c
+++ b/linux-3.0.1/sound/core/control.c
@@ -1419,8 +1419,8 @@ static int snd_ctl_dev_register(struct snd_device *device)
 	if (snd_BUG_ON(cardnum < 0 || cardnum >= SNDRV_CARDS))
 		return -ENXIO;
 	sprintf(name, "controlC%i", cardnum);
-	if ((err = snd_register_device(SNDRV_DEVICE_TYPE_CONTROL, card, -1,
-				       &snd_ctl_f_ops, card, name)) < 0)
+	pr_info("%s. name=%s\n", __func__, name);
+	if ((err = snd_register_device(SNDRV_DEVICE_TYPE_CONTROL, card, -1, &snd_ctl_f_ops, card, name)) < 0)
 		return err;
 	return 0;
 }
diff --git a/linux-3.0.1/sound/core/info.c b/linux-3.0.1/sound/core/info.c
index 5e16adb..525753c 100644
--- a/linux-3.0.1/sound/core/info.c
+++ b/linux-3.0.1/sound/core/info.c
@@ -600,6 +600,8 @@ int snd_info_card_create(struct snd_card *card)
 	char str[8];
 	struct snd_info_entry *entry;
 
+	pr_info("Enter %s\n", __func__);
+
 	if (snd_BUG_ON(!card))
 		return -ENXIO;
 
diff --git a/linux-3.0.1/sound/core/init.c b/linux-3.0.1/sound/core/init.c
index 1e70896..bd9b7b2 100644
--- a/linux-3.0.1/sound/core/init.c
+++ b/linux-3.0.1/sound/core/init.c
@@ -151,7 +151,10 @@ int snd_card_create(int idx, const char *xid, struct module *module, int extra_s
 	struct snd_card *card;
 	int err, idx2;
 
-	printk(KERN_INFO "Enter %s\n", __func__);
+	printk(KERN_INFO "Enter %s ++++++++++++++++++++++\n", __func__);
+	if(card_ret->number && card_ret->id && card_ret->driver && card_ret->shortname)
+		pr_info("%s. card_ret->number=%s && card_ret->id=%s && card_ret->driver=%s && card_ret->shortname=%s\n",
+				__func__, card_ret->number && card_ret->id && card_ret->driver && card_ret->shortname);
 
 	if (snd_BUG_ON(!card_ret))
 		return -EINVAL;
diff --git a/linux-3.0.1/sound/core/pcm.c b/linux-3.0.1/sound/core/pcm.c
index ef01a70..9076df3 100644
--- a/linux-3.0.1/sound/core/pcm.c
+++ b/linux-3.0.1/sound/core/pcm.c
@@ -549,6 +549,7 @@ static int snd_pcm_substream_proc_init(struct snd_pcm_substream *substream)
 	card = substream->pcm->card;
 
 	sprintf(name, "sub%i", substream->number);
+	pr_err("%s. name=%s\n", __func__, name);
 	if ((entry = snd_info_create_card_entry(card, name, substream->pstr->proc_root)) == NULL)
 		return -ENOMEM;
 	entry->mode = S_IFDIR | S_IRUGO | S_IXUGO;
@@ -666,6 +667,7 @@ int snd_pcm_new_stream(struct snd_pcm *pcm, int stream, int substream_count)
 		substream->number = idx;
 		substream->stream = stream;
 		sprintf(substream->name, "subdevice #%i", idx);
+		pr_err("%s. substream->name=%s\n", __func__, substream->name);
 		substream->buffer_bytes_max = UINT_MAX;
 		if (prev == NULL)
 			pstr->substream = substream;
diff --git a/linux-3.0.1/sound/soc/soc-cache.c b/linux-3.0.1/sound/soc/soc-cache.c
index 8c2a21a..d95739f 100644
--- a/linux-3.0.1/sound/soc/soc-cache.c
+++ b/linux-3.0.1/sound/soc/soc-cache.c
@@ -1575,8 +1575,7 @@ int snd_soc_cache_init(struct snd_soc_codec *codec)
 
 	/* Fall back to flat compression */
 	if (i == ARRAY_SIZE(cache_types)) {
-		dev_warn(codec->dev, "Could not match compress type: %d\n",
-			 codec->compress_type);
+		dev_warn(codec->dev, "Could not match compress type: %d\n", codec->compress_type);
 		i = 0;
 	}
 
@@ -1585,8 +1584,8 @@ int snd_soc_cache_init(struct snd_soc_codec *codec)
 
 	if (codec->cache_ops->init) {
 		if (codec->cache_ops->name)
-			dev_dbg(codec->dev, "Initializing %s cache for %s codec\n",
-				codec->cache_ops->name, codec->name);
+			pr_info("Initializing cache=%s for codec=%s. codec->cache_ops->init=>%p\n",
+					codec->cache_ops->name, codec->name, codec->cache_ops->init);
 		return codec->cache_ops->init(codec);
 	}
 	return -EINVAL;
diff --git a/linux-3.0.1/sound/soc/soc-core.c b/linux-3.0.1/sound/soc/soc-core.c
index 0b0e89f..9986a5f 100644
--- a/linux-3.0.1/sound/soc/soc-core.c
+++ b/linux-3.0.1/sound/soc/soc-core.c
@@ -409,8 +409,8 @@ static int soc_ac97_dev_register(struct snd_soc_codec *codec)
 	codec->ac97->dev.parent = codec->card->dev;
 	codec->ac97->dev.release = soc_ac97_device_release;
 
-	dev_set_name(&codec->ac97->dev, "%d-%d:%s",
-		     codec->card->snd_card->number, 0, codec->name);
+	dev_set_name(&codec->ac97->dev, "%d-%d:%s", codec->card->snd_card->number, 0, codec->name);
+	pr_info("%s. name=%d-%d:%s\n", __func__, codec->card->snd_card->number, 0, codec->name);
 	err = device_register(&codec->ac97->dev);
 	if (err < 0) {
 		snd_printk(KERN_ERR "Can't register ac97 bus\n");
@@ -1457,10 +1457,13 @@ static int soc_post_component_init(struct snd_soc_card *card, struct snd_soc_cod
 	codec->name_prefix = NULL;
 
 	/* do machine specific initialization */
-	if (!dailess && dai_link->init)
+	if (!dailess && dai_link->init) {
+		pr_err("%s. name=%s, dai_link->init=>%p\n",__func__,name, dai_link->init);
 		ret = dai_link->init(rtd);
-	else if (dailess && aux_dev->init)
+	} else if (dailess && aux_dev->init) {
+		pr_err("%s. name=%s, aux_dev->init=>%p\n",__func__, name, aux_dev->init);
 		ret = aux_dev->init(&codec->dapm);
+	}
 	if (ret < 0) {
 		dev_err(card->dev, "asoc: failed to init %s: %d\n", name, ret);
 		return ret;
@@ -1519,6 +1522,7 @@ static int soc_probe_dai_link(struct snd_soc_card *card, int num)
 	rtd->pmdown_time = pmdown_time;
 
 	/* probe the cpu_dai */
+	pr_info("%s. probe the cpu_dai +++++++++++++++++\n", __func__);
 	if (!cpu_dai->probed) {
 		if (cpu_dai->driver->probe) {
 			printk(KERN_ERR "%s.cpu_dai->driver->probe=>%p\n", __func__, cpu_dai->driver->probe);
@@ -1534,6 +1538,7 @@ static int soc_probe_dai_link(struct snd_soc_card *card, int num)
 	}
 
 	/* probe the CODEC */
+	pr_info("%s. probe the CODEC +++++++++++++++++\n", __func__);
 	if (!codec->probed) {
 		ret = soc_probe_codec(card, codec);
 		if (ret < 0)
@@ -1541,6 +1546,7 @@ static int soc_probe_dai_link(struct snd_soc_card *card, int num)
 	}
 
 	/* probe the platform */
+	pr_info("%s. probe the platform +++++++++++++++++\n", __func__);
 	if (!platform->probed) {
 		if (platform->driver->probe) {
 			printk(KERN_ERR "%s. platform->driver->probe=>%p\n", __func__, platform->driver->probe);
@@ -1560,6 +1566,7 @@ static int soc_probe_dai_link(struct snd_soc_card *card, int num)
 	}
 
 	/* probe the CODEC DAI */
+	pr_info("%s. probe the CODEC DAI +++++++++++++++++\n", __func__);
 	if (!codec_dai->probed) {
 		if (codec_dai->driver->probe) {
 			printk(KERN_ERR "%s. codec_dai->driver->probe=>%p\n", __func__, codec_dai->driver->probe);
@@ -1576,6 +1583,7 @@ static int soc_probe_dai_link(struct snd_soc_card *card, int num)
 	}
 
 	/* DAPM dai link stream work */
+	pr_info("%s. DAPM dai link stream work +++++++++++++++++\n", __func__);
 	INIT_DELAYED_WORK(&rtd->delayed_work, close_delayed_work);
 
 	ret = soc_post_component_init(card, codec, num, 0);
@@ -1587,6 +1595,7 @@ static int soc_probe_dai_link(struct snd_soc_card *card, int num)
 		printk(KERN_WARNING "asoc: failed to add pmdown_time sysfs\n");
 
 	/* create the pcm */
+	pr_info("%s. create the pcm +++++++++++++++++\n", __func__);
 	ret = soc_new_pcm(rtd, num);
 	if (ret < 0) {
 		printk(KERN_ERR "asoc: can't create pcm %s\n", dai_link->stream_name);
@@ -1594,6 +1603,7 @@ static int soc_probe_dai_link(struct snd_soc_card *card, int num)
 	}
 
 	/* add platform data for AC97 devices */
+	pr_info("%s. add platform data for AC97 devices  +++++++++++++++++\n", __func__);
 	if (rtd->codec_dai->driver->ac97_control)
 		snd_ac97_dev_add_pdata(codec->ac97, rtd->cpu_dai->ac97_pdata);
 
@@ -1702,8 +1712,7 @@ static int snd_soc_init_codec_cache(struct snd_soc_codec *codec,
 		codec->compress_type = compress_type;
 	ret = snd_soc_cache_init(codec);
 	if (ret < 0) {
-		dev_err(codec->dev, "Failed to set cache compression type: %d\n",
-			ret);
+		dev_err(codec->dev, "Failed to set cache compression type: %d\n", ret);
 		return ret;
 	}
 	codec->cache_init = 1;
@@ -2109,7 +2118,7 @@ unsigned int snd_soc_read(struct snd_soc_codec *codec, unsigned int reg)
 	unsigned int ret;
 
 	ret = codec->read(codec, reg);
-	dev_info(codec->dev, "codec %s reads reg=%x => %x\n", codec->name, reg, ret);
+	pr_err("%s. codecName=%s, readsReg=%x, ret=0x%x\n", __func__, codec->name, reg, ret);
 	trace_snd_soc_reg_read(codec, reg, ret);
 
 	return ret;
@@ -3212,8 +3221,7 @@ static inline char *fmt_multiple_name(struct device *dev,
 		struct snd_soc_dai_driver *dai_drv)
 {
 	if (dai_drv->name == NULL) {
-		printk(KERN_ERR "asoc: error - multiple DAI %s registered with no name\n",
-				dev_name(dev));
+		printk(KERN_ERR "asoc: error - multiple DAI %s registered with no name\n", dev_name(dev));
 		return NULL;
 	}
 
diff --git a/linux-3.0.1/sound/soc/soc-dapm.c b/linux-3.0.1/sound/soc/soc-dapm.c
index 3049c54..41f26ea 100644
--- a/linux-3.0.1/sound/soc/soc-dapm.c
+++ b/linux-3.0.1/sound/soc/soc-dapm.c
@@ -997,7 +997,7 @@ static int dapm_power_widgets(struct snd_soc_dapm_context *dapm, int event)
 	int ret = 0;
 	int power;
 
-	printk(KERN_INFO "%s\n", __func__);
+	printk(KERN_INFO "%s +++++++++++++++++++++++++\n", __func__);
 
 	trace_snd_soc_dapm_start(card);
 
@@ -1010,33 +1010,22 @@ static int dapm_power_widgets(struct snd_soc_dapm_context *dapm, int event)
 	 */
 	list_for_each_entry(w, &card->widgets, list) {
 		switch (w->id) {
-		case snd_soc_dapm_pre:
-			dapm_seq_insert(w, &down_list, dapm_down_seq);
-			break;
-		case snd_soc_dapm_post:
-			dapm_seq_insert(w, &up_list, dapm_up_seq);
-			break;
+		case snd_soc_dapm_pre:			dapm_seq_insert(w, &down_list, dapm_down_seq);	break;
+		case snd_soc_dapm_post:			dapm_seq_insert(w, &up_list, dapm_up_seq);	break;
 
 		default:
-			if (!w->power_check)
-				continue;
+			if (!w->power_check)		continue;
 
-			if (!w->force)
-				power = w->power_check(w);
-			else
-				power = 1;
-			if (power)
-				w->dapm->dev_power = 1;
+			if (!w->force)			power = w->power_check(w);
+			else				power = 1;
+			if (power)			w->dapm->dev_power = 1;
 
-			if (w->power == power)
-				continue;
+			if (w->power == power)		continue;
 
 			trace_snd_soc_dapm_widget_power(w, power);
 
-			if (power)
-				dapm_seq_insert(w, &up_list, dapm_up_seq);
-			else
-				dapm_seq_insert(w, &down_list, dapm_down_seq);
+			if (power)			dapm_seq_insert(w, &up_list, dapm_up_seq);
+			else				dapm_seq_insert(w, &down_list, dapm_down_seq);
 
 			w->power = power;
 			break;
@@ -1047,28 +1036,17 @@ static int dapm_power_widgets(struct snd_soc_dapm_context *dapm, int event)
 	if (!dapm->n_widgets) {
 		switch (event) {
 		case SND_SOC_DAPM_STREAM_START:
-		case SND_SOC_DAPM_STREAM_RESUME:
-			dapm->dev_power = 1;
-			break;
-		case SND_SOC_DAPM_STREAM_STOP:
-			dapm->dev_power = !!dapm->codec->active;
-			break;
-		case SND_SOC_DAPM_STREAM_SUSPEND:
-			dapm->dev_power = 0;
-			break;
+		case SND_SOC_DAPM_STREAM_RESUME:	dapm->dev_power = 1;				break;
+		case SND_SOC_DAPM_STREAM_STOP:		dapm->dev_power = !!dapm->codec->active;	break;
+		case SND_SOC_DAPM_STREAM_SUSPEND:	dapm->dev_power = 0;				break;
 		case SND_SOC_DAPM_STREAM_NOP:
 			switch (dapm->bias_level) {
 				case SND_SOC_BIAS_STANDBY:
-				case SND_SOC_BIAS_OFF:
-					dapm->dev_power = 0;
-					break;
-				default:
-					dapm->dev_power = 1;
-					break;
+				case SND_SOC_BIAS_OFF:		dapm->dev_power = 0;			break;
+				default:			dapm->dev_power = 1;			break;
 			}
 			break;
-		default:
-			break;
+		default:										break;
 		}
 	}
 
@@ -1084,17 +1062,18 @@ static int dapm_power_widgets(struct snd_soc_dapm_context *dapm, int event)
 		    (!d->dev_power && d->bias_level == SND_SOC_BIAS_ON)) {
 			ret = snd_soc_dapm_set_bias_level(card, d, SND_SOC_BIAS_PREPARE);
 			if (ret != 0)
-				dev_err(d->dev,
-					"Failed to prepare bias: %d\n", ret);
+				dev_err(d->dev, "Failed to prepare bias: %d\n", ret);
 		}
 	}
 
 	/* Power down widgets first; try to avoid amplifying pops. */
+	printk(KERN_INFO "%s. Power down widgets first\n", __func__);
 	dapm_seq_run(dapm, &down_list, event, dapm_down_seq);
 
 	dapm_widget_update(dapm);
 
 	/* Now power up. */
+	printk(KERN_INFO "%s. Now power up ++++++++++++\n", __func__);
 	dapm_seq_run(dapm, &up_list, event, dapm_up_seq);
 
 	list_for_each_entry(d, &dapm->card->dapm_list, list) {
@@ -1622,49 +1601,45 @@ int snd_soc_dapm_new_widgets(struct snd_soc_dapm_context *dapm)
 {
 	struct snd_soc_dapm_widget *w;
 	unsigned int val;
+
 	printk(KERN_ERR "%s\n", __func__);
+
 	list_for_each_entry(w, &dapm->card->widgets, list)
 	{
 		if (w->new)
 			continue;
 
+		pr_err("w->id=%d\n", w->id);
+
 		switch(w->id) {
 		case snd_soc_dapm_switch:
 		case snd_soc_dapm_mixer:
-		case snd_soc_dapm_mixer_named_ctl:
-			w->power_check = dapm_generic_check_power;
-			dapm_new_mixer(dapm, w);
-			break;
+		case snd_soc_dapm_mixer_named_ctl:  w->power_check = dapm_generic_check_power;
+						dapm_new_mixer(dapm, w);
+						break;
 		case snd_soc_dapm_mux:
 		case snd_soc_dapm_virt_mux:
-		case snd_soc_dapm_value_mux:
-			w->power_check = dapm_generic_check_power;
-			dapm_new_mux(dapm, w);
-			break;
+		case snd_soc_dapm_value_mux:   	w->power_check = dapm_generic_check_power;
+						dapm_new_mux(dapm, w);
+						break;
 		case snd_soc_dapm_adc:
-		case snd_soc_dapm_aif_out:
-			w->power_check = dapm_adc_check_power;
-			break;
+		case snd_soc_dapm_aif_out:   	w->power_check = dapm_adc_check_power;
+						break;
 		case snd_soc_dapm_dac:
-		case snd_soc_dapm_aif_in:
-			w->power_check = dapm_dac_check_power;
-			break;
+		case snd_soc_dapm_aif_in:   	w->power_check = dapm_dac_check_power;
+						break;
 		case snd_soc_dapm_pga:
-		case snd_soc_dapm_out_drv:
-			w->power_check = dapm_generic_check_power;
-			dapm_new_pga(dapm, w);
-			break;
+		case snd_soc_dapm_out_drv:   	w->power_check = dapm_generic_check_power;
+						dapm_new_pga(dapm, w);
+						break;
 		case snd_soc_dapm_input:
 		case snd_soc_dapm_output:
 		case snd_soc_dapm_micbias:
 		case snd_soc_dapm_spk:
 		case snd_soc_dapm_hp:
 		case snd_soc_dapm_mic:
-		case snd_soc_dapm_line:
-			w->power_check = dapm_generic_check_power;
-			break;
-		case snd_soc_dapm_supply:
-			w->power_check = dapm_supply_check_power;
+		case snd_soc_dapm_line:		w->power_check = dapm_generic_check_power;	break;
+		case snd_soc_dapm_supply:	w->power_check = dapm_supply_check_power;  	/* No break ? */
 		case snd_soc_dapm_vmid:
 		case snd_soc_dapm_pre:
 		case snd_soc_dapm_post:
@@ -1675,11 +1650,9 @@ int snd_soc_dapm_new_widgets(struct snd_soc_dapm_context *dapm)
 		if (w->reg >= 0) {
 			val = snd_soc_read(w->codec, w->reg);
 			val &= 1 << w->shift;
-			if (w->invert)
-				val = !val;
+			if (w->invert)		val = !val;
 
-			if (val)
-				w->power = 1;
+			if (val)		w->power = 1;
 		}
 
 		w->new = 1;
